EpikCord/client/http_client.py

```diff
@@ -1,312 +1,233 @@
-            if _ORJSON
-            else json.dumps(x),
-            ws_response_class=GatewayWebsocket,
-            headers=headers,
+    def __init__(self, token: str, *, version: int = 10):
+        self.token: str = token
+        self.version: int = version
+        self.session: aiohttp.ClientSession = aiohttp.ClientSession(
+            headers={"Authorization": f"Bot {self.token}"}
         )
-
-        self.ws_connect = self.session.ws_connect
-
-        self.global_ratelimit: asyncio.Event = asyncio.Event()
-        self.global_ratelimit.set()
         self.buckets: Dict[str, Bucket] = {}
+        self.global_event: asyncio.Event = asyncio.Event()
+        self.global_event.set()
+
+    async def ws_connect(self, url: str) -> aiohttp.ClientWebSocketResponse:
+        return await self.session.ws_connect(url)
 
     async def request(
         self,
-        method,
-        url,
+        method: str,
+        url: str,
         *args,
-        attempt: int = 1,
-        to_discord=True,
-        guild_id: Union[str, int] = 0,
-        channel_id: Union[int, str] = 0,
-        reason: Optional[str] = None,
+        discord: bool = True,
+        channel_id: Optional[int] = None,
+        guild_id: Optional[int] = None,
+        webhook_id: Optional[int] = None,
+        webhook_token: Optional[str] = None,
         **kwargs,
-    ):
-        if attempt > 5:
-            logger.critical(f"Failed a {method} {url} 5 times.")
-            return
+    ) -> Optional[aiohttp.ClientResponse]:
+        """
+        Parameters
+        ----------
+        method : str
+            The method of the request.
+        url : str
+            The url of the request.
+        *args
+            The args to pass to the ClientSession.request method.
+        discord : bool
+            Whether this request is to Discord.
+        channel_id : int
+            The channel id of the request.
+        guild_id : int
+            The guild id of the request.
+        webhook_id : int
+            The webhook id of the request.
+        webhook_token : str
+            The webhook token of the request.
+        **kwargs
+            The kwargs to pass to the ClientSession.request method.
+
+        Returns
+        -------
+        Optional[aiohttp.ClientResponse]
+            The response of the request. Can be None if the request fails.
+
+        Raises
+        ------
+        NotFound
+            If the request returns a 404 status code.
+        Forbidden
+            If the request returns a 403 status code.
+        Unauthorized
+            If the request returns a 401 status code.
+        BadRequest
+            If the request returns a 400 status code.
+        HTTPException
+            If the request returns a status code that is not OK and not any of the other exceptions.
+        """
 
-        if url.startswith("ws") or not to_discord:
+        if not discord:
             return await self.session.request(method, url, *args, **kwargs)
```

Comment:
I feel like these non-discord requests should be apart of a separate function. But I also don't see much use in even having such a function.

Reply:
I mean it creates a way for you to not have 2 `ClientSessions`, by using a single kwarg, and I think that's a nice thing.